### PR TITLE
Add section for modulecalls

### PR DESCRIPTION
**Pull request by HeikoNeblung**
### Prerequisites

Put an `x` into the box(es) that apply:

- [ ] This pull request fixes a bug.
- [x] This pull request adds a feature.
- [x] This pull request enhances existing functionality.
- [ ] This pull request introduces breaking change.

For more information, see the [Contributing Guide](https://github.com/terraform-docs/terraform-docs/tree/master/CONTRIBUTING.md).

### Description

This PR adds a section for modulecalls including name, source and version of the modulecall.

### Issues Resolved

* #365

### Checklist

Put an `x` into all boxes that apply:

- [x] I have read the [Contributing Guidelines](https://github.com/terraform-docs/terraform-docs/tree/master/CONTRIBUTING.md).

#### Tests

- [x] I have added tests to cover my changes.
- [x] All tests pass when I run `make test`.

#### Documentation

- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

#### Code Style

- [x] My code follows the code style of this project.

### CODE REVIEW

**Review by khos2ow — CHANGES_REQUESTED** [review]
@HeikoNeblung thank you for this, but you should not rename the imports in the project due to forking. My guess is that you've cloned this locally at `$GOPATH/src/github/heikoneblung` which is wrong. You should move the project, locally on your machine, to `$GOPATH/src/github/terraform-docs/terraform-docs` and then add an extra git remote to be able.

Please revert all the import renaming (this commit https://github.com/terraform-docs/terraform-docs/pull/367/commits/eb7e37f4cda3f1cf314aeb516e04b9b64679b217) for us to be able to proceed with reviewing and testing your PR.

**Review by khos2ow — COMMENTED** [review]
Thank you @HeikoNeblung for the work so far.

At a glance this seems to be ok so far, but since we're in the middle of reviewing another PR (#347) which touches almost all the files here I would like to wait for that to be finished and merged and then pick this up for review. Otherwise you'd need to rebase and fix conflicts multiple times.

**Comment by khos2ow** [comment]
@HeikoNeblung would you be able to rebase this on master and fix the conflicts?

**Comment by HeikoNeblung** [comment]
> @HeikoNeblung would you be able to rebase this on master and fix the conflicts?

@khos2ow All tests are passing locally on my machine. I could provide PR for plugin-sdk containing my local changes if you like.

**Comment by khos2ow** [comment]
xref: https://github.com/terraform-docs/plugin-sdk/pull/10

File: cmd/root.go [hunk 1]
```diff
@@ -57,8 +57,8 @@ func NewCommand() *cobra.Command {
 	// flags
 	cmd.PersistentFlags().StringVarP(&config.File, "config", "c", ".terraform-docs.yml", "config file name")
 
-	cmd.PersistentFlags().StringSliceVar(&config.Sections.Show, "show", []string{}, "show section [header, inputs, outputs, providers, requirements, resources]")
-	cmd.PersistentFlags().StringSliceVar(&config.Sections.Hide, "hide", []string{}, "hide section [header, inputs, outputs, providers, requirements, resources]")
+	cmd.PersistentFlags().StringSliceVar(&config.Sections.Show, "show", []string{}, "show section [header, inputs, modulecalls, outputs, providers, requirements, resources]")
```
Comment (khos2ow) [inline cmd/root.go:60]: Please rename user-facing `modulecalls` to simply `modules`. Note that we should still keep `modulecalls` for internal variable and struct names.

File: internal/format/asciidoc_document.go [hunk 2]
```diff
@@ -149,13 +149,33 @@ const (
...
+	`
+
 	asciidocDocumentTpl = `
 	{{- template "header" . -}}
 	{{- template "requirements" . -}}
 	{{- template "providers" . -}}
 	{{- template "resources" . -}}
 	{{- template "inputs" . -}}
 	{{- template "outputs" . -}}
+	{{- template "modulecalls" . -}}
```
Comment (khos2ow) [inline internal/format/asciidoc_document.go:178]: Please move modules section right after providers section. This applies to asciidoc, markdown, and pretty.

File: internal/format/markdown_document.go [hunk 3]
```diff
@@ -149,13 +149,34 @@ const (
 	{{ end -}}
 	`
 
+	documentModulecallsTpl = `
+	{{- if .Settings.ShowModulecalls -}}
+		{{ indent 0 "#" }} Modulecalls
+		{{ if not .Module.Modulecalls }}
+			No Modulecalls.
+		{{ else }}
+			the following Modules are called:
```
Comment (khos2ow) [inline internal/format/markdown_document.go:158]:
```suggestion
			The following Modules are called:
```

File: internal/format/asciidoc_document.go [hunk 4]
```diff
@@ -149,13 +149,33 @@ const (
 	{{ end -}}
 	`
 
+	asciidocDocumentModulecallsTpl = `
+	{{- if .Settings.ShowModulecalls -}}
+		{{ indent 0 "=" }} Modulecalls
+		{{ if not .Module.Modulecalls }}
+			No Modulecalls.
+		{{ else }}
+			the following Modules are called:
```
Comment (khos2ow) [inline internal/format/asciidoc_document.go:158]:
```suggestion
			The following Modules are called:
```

**Review by khos2ow — COMMENTED** [review]
Thanks @HeikoNeblung, overall this looks great with couple comments below. Also I think it should be better to use full camel case `ModuleCalls` rather than `Modulecalls`.

File: internal/format/asciidoc_document.go [hunk 5]
```diff
@@ -149,10 +149,30 @@ const (
 	{{ end -}}
 	`
 
+	asciidocDocumentModulecallsTpl = `
+	{{- if .Settings.ShowModuleCalls -}}
+		{{ indent 0 "=" }} Modulecalls
```
Comment (khos2ow) [inline internal/format/asciidoc_document.go:154]: Rename to `Modules`

File: internal/format/asciidoc_document.go [hunk 6]
```diff
@@ -149,10 +149,30 @@ const (
...
+		{{ if not .Module.ModuleCalls }}
+			No Modulecalls.
```
Comment (khos2ow) [inline internal/format/asciidoc_document.go:156]: Same rename.

File: internal/format/asciidoc_table.go [hunk 7]
```diff
@@ -127,10 +127,28 @@ const (
 	{{ end -}}
 	`
 
+	asciidocTableModulecallsTpl = `
+	{{- if .Settings.ShowModuleCalls -}}
+		{{ indent 0 "=" }} Modulecalls
```
Comment (khos2ow) [inline internal/format/asciidoc_table.go:132]: Rename.

File: internal/format/markdown_document.go [hunk 8]
```diff
@@ -149,10 +149,31 @@ const (
 	{{ end -}}
 	`
 
+	documentModulecallsTpl = `
+	{{- if .Settings.ShowModuleCalls -}}
+		{{ indent 0 "#" }} Modulecalls
```
Comment (khos2ow) [inline internal/format/markdown_document.go:154]: Rename.

File: internal/format/markdown_table.go [hunk 9]
```diff
@@ -116,10 +116,26 @@ const (
 	{{ end -}}
 	`
 
+	tableModulecallsTpl = `
+	{{- if .Settings.ShowModuleCalls -}}
+		{{ indent 0 "#" }} Modulecalls
```
Comment (khos2ow) [inline internal/format/markdown_table.go:121]: Rename.

File: internal/terraform/module.go [hunk 10]
```diff
@@ -40,6 +41,7 @@ type Module struct {
 
 	Header       string         `json:"header" toml:"header" xml:"header" yaml:"header"`
 	Inputs       []*Input       `json:"inputs" toml:"inputs" xml:"inputs>input" yaml:"inputs"`
+	ModuleCalls  []*ModuleCall  `json:"modulecalls" toml:"modulecalls" xml:"modulecalls>modulecall" xaml:"modulecalls"`
```
Comment (khos2ow) [inline internal/terraform/module.go:44]:
```suggestion
	ModuleCalls  []*ModuleCall  `json:"modules" toml:"modules" xml:"modules>module" yaml:"modules"`
```

**Review by khos2ow — COMMENTED** [review]
Some comments below of renaming nature, otherwise looks good!

**Comment by khos2ow** [comment]
Also I just noticed all the required changes in `internal/cli/config.go` are missing. Would you mind taking a look at that file as well?

File: internal/format/asciidoc_document.go (hunk 5 above)
Comment (HeikoNeblung) [reply internal/format/asciidoc_document.go]: renamed

File: internal/format/asciidoc_document.go (hunk 6 above)
Comment (HeikoNeblung) [reply internal/format/asciidoc_document.go]: renamed

File: internal/format/asciidoc_table.go (hunk 7 above)
Comment (HeikoNeblung) [reply internal/format/asciidoc_table.go]: renamed

File: internal/format/markdown_document.go (hunk 8 above)
Comment (HeikoNeblung) [reply internal/format/markdown_document.go]: renamed

File: internal/format/markdown_table.go (hunk 9 above)
Comment (HeikoNeblung) [reply internal/format/markdown_table.go]: ranamed

File: internal/terraform/module.go (hunk 10 above)
Comment (HeikoNeblung) [reply internal/terraform/module.go]: renamed

**Comment by HeikoNeblung** [comment]
I did not squash my  commit to let you review my changes more easy. Please let me know when I should do so.

File: examples/.terraform-docs.yml [hunk 11]
```diff
@@ -6,6 +6,7 @@ sections:
     - header
     - inputs
     - providers
+    - modulecalls
```
Comment (khos2ow) [inline examples/.terraform-docs.yml:9]: Rename.

**Review by khos2ow — COMMENTED** [review]
One last rename (see below), also please go ahead and add `modules` to `docs/CONFIG_FILE.md` for `sections/hide` and `sections/show` (for example purpose). Then please go ahead and squash! :+1:

**Review by khos2ow — APPROVED** [review]
Thank you @HeikoNeblung, LGTM!

**Pull request merged**